app/models/organisation.rb

```diff
@@ -193,10 +193,8 @@ def foi_contacts
 
   before_destroy { |r| throw :abort unless r.destroyable? }
   after_save :ensure_analytics_identifier
-
-  # We only update links when a new organisation is created, since
-  # dependency resolution will take care of all updates and deletes.
-  after_create :update_organisations_index_page_links
+  after_save :update_organisations_index_page_links
+  after_destroy :update_organisations_index_page_links
```

Comment:
We should probably rename this method now if it's no longer links

Reply:
Also, we should be consistent over whether it's the home page or the index page.  I think I prefer index page which means renaming `PublishOrganisationsHomepage`, but I'm not really fussed which way round we go.

Reply:
I've renamed everything to refer to the index page to reduce confusion with individual organisation homepages